src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+                            var indexer = from p in allMembers.OfType<IPropertySymbol>()
+                                          where p.IsIndexer
+                                          select SymbolCompletionItem.CreateWithSymbolId(
+                                              displayText: $"[{string.Join(", ", p.Parameters.Select(p => p.Type.ToMinimalDisplayString(semanticModel, position)))}]", // The type to convert to
+                                              symbols: ImmutableList.Create(p),
+                                              rules: CompletionItemRules.Default,
+                                              contextPosition: position,
+                                              properties: CreateCompletionHandlerProperty(CompletionHandlerIndexer));
+                            context.AddItems(allExplicitConversions.Union(indexer));
+                        }
+                    }
+                }
+            }
+        }
+
+        private static ExpressionSyntax? GetExpressionOfInvocation(SyntaxToken dotToken)
+        {
+            Debug.Assert(dotToken.IsKind(SyntaxKind.DotToken));
+            return dotToken.Parent switch
+            {
+                MemberAccessExpressionSyntax memberAccess => memberAccess.Expression,
+                MemberBindingExpressionSyntax { Parent: ConditionalAccessExpressionSyntax conditionalAccess } _ => conditionalAccess.Expression,
+                _ => null,
+            };
+        }
+
+        internal override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, TextSpan completionListSpan, char? commitKey, bool disallowAddingImports, CancellationToken cancellationToken)
+        {
+            if (item.Properties.TryGetValue(CompletionHandlerPropertyname, out var value))
+            {
+                var completionChange = value switch
+                {
+                    CompletionHandlerConversion => await HandleConversionChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    CompletionHandlerIndexer => await HandleIndexerChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    _ => null,
+                };
+                if (completionChange is { })
+                {
+                    return completionChange;
+                }
+            }
+
+            return await base.GetChangeAsync(document, item, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<CompletionChange?> HandleConversionChangeAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var symbols = await SymbolCompletionItem.GetSymbolsAsync(item, document, cancellationToken).ConfigureAwait(false);
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            var symbol = symbols.Single() as IMethodSymbol;
```

Comment:
i don't trust that you know you'll only get a .Single result when resolving the item.  indeed, in error conditions (multiple indexers/conversions with the same sig), i think you could get multiple.

Reply:
This is the conversion case and we need to suggest every conversion as its own entry, so I think it's safe. At least that is the way I created the completion item: `symbols: ImmutableList.Create(m)`

Reply:
it's not safe :)  the way this works is that the symbol is encoded as a string and stored in a property bag.  we then have a resolution step that tries to map back from string->symbol.  However:

1. that is not guaranteed to succeed.
2. that is not guaranteed to give you a single symbol back.